snow/engine/snowman/block/vm.go

```diff
@@ -42,6 +42,12 @@ type ChainVM interface {
 	// This should always be a block that has no children known to consensus.
 	SetPreference(ctx context.Context, blkID ids.ID) error
 
+	// GetPreference returns the ID of the currently preferred block.
+	// If no blocks have been accepted/preferred by consensus yet, it is
+	// assumed there is a definitionally accepted block, the Genesis block, that
+	// will be returned.
+	GetPreference(context.Context) (ids.ID, error)
```

Comment:
Chatted w/ @StephenButtolph  + @aaronbuchwald  offline, since it's strange that this is added to the VM interface if proposervm is always handling preference persistence, any implementation in the underlying VM is pretty much overridden. @StephenButtolph  suggested that we should bubble down the proposervm so it's exposed as a library that `vm.Factory#New` needs to use to wrap its VM to add congestion control support as an opt-in feature.

Reply:
Bubbling this down sounds like a great change